### ISSUE TITLE
RestClient fetches data before Auth

### ISSUE THREAD

**Issue by kidphys** [sidferreira/aor-firebase-client#21]
When I try to add RestClient to admin like this:
```
 <Admin restClient={RestClient(trackedResources, firebaseConfig)} authClient={AuthClient}>
      <Resource name="users" list={UserList} edit={UserEdit} icon={UserIcon}/>
 </Admin>
```
And the resource "users" is guarded by firebase permission rules, and is not loaded. The login page is either hanged if access "/users" directly, or login successfully but then "users" is not loaded.

After debugging, it seems to me that the firebase client is binding to the resource "users" without checking for authentication status, the firebase web-socket frame returns "permission denied" and then hang then.

There might be 2 improvements needed:
- Making sure firebase authentication status is checked before binding to resource.
- Allow 'refresh' action to fetch the resource again, may be re-init the firebase connection is required.

**Comment by sidferreira** [comment]
@kidphys Great input! Any chance you can work on that and help us?

**Comment by sidferreira** [comment]
@kidphys FIxed on v0.2.0
